test/uk/gov/hmrc/helptosavefrontend/repo/MongoTestSupport.scala

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.repo
+
+import org.scalamock.scalatest.MockFactory
+import play.api.libs.json.{JsString, Json}
+import play.api.libs.json.Json.toJsFieldJsValueWrapper
+import play.modules.reactivemongo.ReactiveMongoComponent
+import reactivemongo.api.DefaultDB
+import reactivemongo.bson.BSONObjectID
+import uk.gov.hmrc.mongo.{MongoConnector, ReactiveRepository}
+
+import scala.concurrent.Future
+
+trait MongoTestSupport[Data, R <: ReactiveRepository[Data,BSONObjectID]] { this: MockFactory ⇒
+
+  trait MockDBFunctions {
```

Comment:
if we search in gihub for `MongoTestSupport` we see how others are mocking mongo stuff they dont use `MockDBFunctions`